### PR TITLE
Add Linear Regression algorithm

### LINKED ISSUE #5: MachinelLearning_Toolkit: Is there a linear regression function in the toolkit?
Just wondering if there is such a function existing. If not, can we have one?

### PR DESCRIPTION
**Pull request by epignatelli**
### Issues addressed by this PR

Closes #5 

### Test files

https://burohappold.sharepoint.com/:f:/s/BHoM/EmYIW2WfjUtPkIYccZ-8V8gBZTOmUfLfgK1gllazCEilcQ?e=C174yT

### Changelog

- Add Linear Regression algorithm
- Refactor the way modules are loaded
- Make sure that the `Invoke` method is always passing Python arguments

### CODE REVIEW

**Comment by epignatelli** [comment]
@MarthaVoul @FraserGreenroyd FYI, this can be an example of a code development workflow for adding an ML algorithm!

File: MachineLearning_Engine/MachineLearning_Engine.csproj [hunk 1]
```diff
@@ -48,6 +48,10 @@
       <HintPath>..\..\Python_Toolkit\Build\Python_Engine.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="Reflection_oM, Version=3.0.0.0, Culture=neutral, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\BHoM\Build\Reflection_oM.dll</HintPath>
+    </Reference>
```
Comment (FraserGreenroyd) [inline MachineLearning_Engine/MachineLearning_Engine.csproj]: Reference should not have version, culture, or processorArchitecture.
Should also be set to copy local false :smile:

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Review by BingWangUS — DISMISSED** [review]
Function wise working great!

**Comment by BingWangUS** [comment]
Would like to see an example to see how to structure a data structure in grasshopper for multi-input, x1, x2, x3. 

Also want to have the capacity to expose index from the model.

**Review by FraserGreenroyd — APPROVED** [review]

**Comment by al-fisher** [comment]
> fix installation of ML_Tooltki python side

👍👍 thanks @epignatelli

**Pull request merged**